apps/server/src/AppService.ts

```diff
@@ -30,36 +30,53 @@ export class AppService {
     return this.ethersRpcProvider.getBalance(address);
   }
 
-  async getAllEventsFromBlockNumber(): Promise<Event[]> {
+  async checkTransactionConfirmationStatus(transactionHash: string): Promise<boolean> {
+    const receipt = await this.ethersRpcProvider.getTransactionReceipt(transactionHash);
+    const confirmed = receipt.status === 1;
+    if (!confirmed) {
```

Comment:
```suggestion
    const isReverted = receipt.status === 0;
    if (isReverted === true) {
```

I would prefer explicit equality checks, but that's just a personal nit of mine.